src/chat-model.ts

```diff
@@ -623,23 +629,45 @@ ${toolsList}
     }
 
     try {
-      const model = await this.input.documentManager?.services.contents.get(
+      // Try reading from live notebook if open
+      const widget = this.input.documentManager?.findWidget(
         attachment.value
-      );
-      if (!model || model.type !== 'notebook') {
-        return null;
-      }
+      ) as IDocumentWidget<Notebook, INotebookModel> | undefined;
+      let cellData: nbformat.ICell[] | null = null;
+      let kernelLang = 'text';
+
+      const ymodel = widget?.context?.model?.sharedModel as YNotebook;
+
+      if (ymodel) {
+        const nb = ymodel.toJSON();
+
+        cellData = nb.cells;
 
-      const kernelLang =
-        model.content?.metadata?.language_info?.name ||
-        model.content?.metadata?.kernelspec?.language ||
-        'text';
+        const lang =
+          nb.metadata?.language_info?.name ||
+          nb.metadata?.kernelspec?.language ||
+          'text';
+
+        kernelLang = String(lang);
+      } else {
+        // Fallback: reading from disk
+        const model = await this.input.documentManager?.services.contents.get(
+          attachment.value
+        );
+        if (!model || model.type !== 'notebook') {
+          return null;
+        }
+        cellData = model.content.cells;
+
+        kernelLang =
+          model.content?.metadata?.language_info?.name ||
+          model.content?.metadata?.kernelspec?.language ||
+          'text';
+      }
 
       const selectedCells = attachment.cells
         .map(cellInfo => {
-          const cell = model.content.cells.find(
-            (c: any) => c.id === cellInfo.id
-          );
+          const cell = cellData!.find(c => c.id === cellInfo.id);
```

Comment:
```suggestion
          const cell = cellData.find(c => c.id === cellInfo.id);
```

*Related to above comments*